pkg/cache/scheduler/tas_flavor_snapshot.go

```diff
@@ -558,9 +562,19 @@ func (s *TASFlavorSnapshot) requiredReplacementDomain(tr *TASPodSetRequests, ta
 		return ""
 	}
 	required := isRequired(tr.PodSet.TopologyRequest)
-	if !required {
+	if !required && !slicesConfigured(tr.PodSet.TopologyRequest) {
 		return ""
 	}
+	targetLevel := levelIdx
+	if slicesConfigured(tr.PodSet.TopologyRequest) && tr.Count < *tr.PodSet.TopologyRequest.PodSetSliceSize {
+		sliceTopologyKey := s.sliceLevelKeyWithDefault(tr.PodSet.TopologyRequest, s.lowestLevel())
+		sliceLevelIdx, resolved := s.resolveLevelIdx(sliceTopologyKey)
+		if !resolved {
+			return ""
+		}
+		targetLevel = max(targetLevel, sliceLevelIdx)
```

Comment:
No need to max(levelIdx, sliceLevelIdx), we assume slice will be below required/preferred topologically. If that's not the case, then a different place will return an error